CRM/Contribute/BAO/Contribution.php

```diff
@@ -3122,6 +3122,13 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
         }
       }
     }
+    if ($context == 'changeFinancialType') {
+      foreach ($params['line_item'] as &$lineItems) {
+        foreach ($lineItems as &$line) {
+          $line['financial_type_id'] = $params['financial_type_id'];
+        }
+      }
+    }
   }
 
```

Comment:
Maybe I'm missing something, but the block above doesn't seem to be actually calling any function to update the line_item(s). In any case, I'm not seeing a change to line_item.financial_type_id when I change it in contribution edit form.